crates/sui-types/src/transaction.rs

```diff
@@ -2232,6 +2294,82 @@ impl TransactionDataAPI for TransactionDataV1 {
         self.kind.receiving_objects()
     }
 
+    fn balance_withdraws(&self) -> UserInputResult<BTreeMap<ObjectID, Reservation>> {
```

Comment:
just to make sure I understand - we support multiple reservations for the same ObjectID, but we merge them all down in the return value here so that the scheduler only has to look at the totals for each currency?

Reply:
Correct. Maybe I should rename this function so that it is clear that it should be called by the scheduler. Move would still be taking individual reservations and convert them into individual withdrawal token

Reply:
maybe just `merged_balance_withdraws` or `cumulative_balance_withdraws`?